packages/gestalt/src/Sheet.js

```diff
@@ -51,6 +51,7 @@ type SheetMainProps = {|
   closeOnOutsideClick?: boolean,
   footer?: Node,
   onDismiss: () => void,
+  shouldCloseOnKeyEvent?: () => boolean,
```

Comment:
Does this need to be a function, or could it be a boolean, which defaults to true, and any additional functionality needed can be part of the `onDismiss` passed in?

Reply:
I believe it will be better to use a function in this case, because the code is using "window.addEventListener" directly, without react synthetic events.